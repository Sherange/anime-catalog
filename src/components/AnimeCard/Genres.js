import React from 'react';
import {Text, StyleSheet} from 'react-native';
import {secondryLabel} from '../../constans/theme';
import PropTypes from 'prop-types';

const Genres = ({genres}) => {
  if (genres && genres.length > 0) {
    return genres.map((item, index) => {
      if (index === genres.length - 1) {
        return (
          <Text key={index} style={styles.genresStyle}>
            {item.name}
          </Text>
        );
      } else {
        return (
          <Text key={index} style={styles.genresStyle}>
            {item.name} {'/ '}
          </Text>
        );
      }
    });
  }
};

Genres.propTypes = {
  genres: PropTypes.array.isRequired,
};

const styles = StyleSheet.create({
  genresRow: {
    flexDirection: 'row',
  },
  genresStyle: {
    fontSize: 14,
    color: secondryLabel,
  },
});

export default Genres;
