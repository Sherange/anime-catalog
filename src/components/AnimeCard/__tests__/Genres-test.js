import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Genres from '../Genres';

const mockData = [
  {
    mal_id: 1,
    type: 'anime',
    name: 'Action',
    url: 'https://myanimelist.net/anime/genre/1/Action',
  },
  {
    mal_id: 24,
    type: 'anime',
    name: 'Sci-Fi',
    url: 'https://myanimelist.net/anime/genre/24/Sci-Fi',
  },
];

it('Genres renders correctly', () => {
  renderer.create(<Genres genres={mockData} />);
});
