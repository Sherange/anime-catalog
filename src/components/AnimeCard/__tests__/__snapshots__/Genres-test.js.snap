// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genres snapshot 1`] = `
Array [
  <Text
    style={
      Object {
        "color": "#8A8A8E",
        "fontSize": 12,
      }
    }
  >
    Action
     
    / 
  </Text>,
  <Text
    style={
      Object {
        "color": "#8A8A8E",
        "fontSize": 12,
      }
    }
  >
    Sci-Fi
  </Text>,
]
`;
